### PR TITLE
Add usefull string funcs for tmpl generation

### PR DESCRIPTION
**Pull request by unkeep**

### CODE REVIEW

**Comment by unkeep** [comment]
@hexdigest, hi, could you please merge this?

File: cmd_generate.go [hunk 1]
```diff
@@ -257,6 +257,39 @@ func varsToArgs(v vars) string {
...
+	"upFirst": upFirst,
+	"downFirst": downFirst,
+	"replace": strings.ReplaceAll,
+}
+
+func upFirst(s string) string {
+	var res string
+
+	// iterate over runes
+	for i, c := range s {
```
Comment (hexdigest) [inline cmd_generate.go:269]:
```go
package main

import (
        "strings"
        "testing"
        "unicode"
)

func iterative(s string) string {
        var res string

        // iterate over runes
        for i, c := range s {
                if i == 0 {
                        res += strings.ToLower(string(c))
                } else {
                        res += string(c)
                }
        }

        return res
}

func slicing(s string) string {
        for i, v := range s {
                return string(unicode.ToLower(v)) + s[i+1:]
        }
        return ""
}

func BenchmarkIteratinve(b *testing.B) {
        for i := 0; i < b.N; i++ {
                iterative("Hello world")
        }
}

func BenchmarkSlicing(b *testing.B) {
        for i := 0; i < b.N; i++ {
                slicing("Hello world")
        }
}

```

go test -bench=.
goos: darwin
goarch: amd64
BenchmarkIteratinve-16           2227485               505 ns/op
BenchmarkSlicing-16             26628272                42.0 ns/op
PASS

**Review by hexdigest — CHANGES_REQUESTED** [review]

File: cmd_generate.go [hunk 2]
```diff
@@ -257,6 +257,39 @@ func varsToArgs(v vars) string {
 var helperFuncs = template.FuncMap{
 	"up":   strings.ToUpper,
 	"down": strings.ToLower,
+	"upFirst": upFirst,
+	"downFirst": downFirst,
+	"replace": strings.ReplaceAll,
+}
+
+func upFirst(s string) string {
```
[the remark this answers is not in the document]
Comment (unkeep) [reply cmd_generate.go:266]: mostly, to be consistent to `downFirst`. But, also, `strings.Title` does much more than needed

File: cmd_generate.go [hunk 3]
```diff
@@ -257,6 +257,39 @@ func varsToArgs(v vars) string {
 var helperFuncs = template.FuncMap{
 	"up":   strings.ToUpper,
 	"down": strings.ToLower,
+	"upFirst": upFirst,
+	"downFirst": downFirst,
+	"replace": strings.ReplaceAll,
```
[the remark this answers is not in the document]
Comment (unkeep) [reply cmd_generate.go:263]: upFirst/downFirst - to export/hide an identifier
replace - in my particular case, to remove extra suffix from a type name

File: cmd_generate.go (hunk 1 above)
Comment (unkeep) [reply cmd_generate.go]: `s[i+1:]` will not work for some unicode values, but I took your code as example, now it looks much better, thanks

**Pull request merged**